### PR TITLE
configure mypy outside of pre-commit environment

### PR DESCRIPTION
**Pull request by incognitojam**
- [x] matching configuration in `mypy.ini`
- [x] add types to pipenv dev dependencies (e.g. `types-PyYAML`, `types-pycurl`...)

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Why add the dependencies to the Pipfile? It's not great to have them defined in two places.

**Comment by incognitojam** [comment]
If you configure mypy linting inside an editor it doesn't have the dependencies

**Comment by adeebshihadeh** [comment]
Then let's figure out how to define them once and use in both places.

File: mypy.ini [hunk 1]
```diff
@@ -1,4 +1,14 @@
 [mypy]
 python_version = 3.8
-ignore_missing_imports = True
 plugins = numpy.typing.mypy_plugin
+
+; third-party packages
+ignore_missing_imports = True
```
Comment (adeebshihadeh) [inline mypy.ini:6]: is there necessary outside of pre-commit?

File: mypy.ini [hunk 2]
```diff
@@ -1,4 +1,14 @@
 [mypy]
```
Comment (adeebshihadeh) [inline mypy.ini:1]: make sure to remove all the configuration in pre-commit and point it to this config file if necessary

Comment (incognitojam) [reply mypy.ini:1]: we might have to change the pre-commit hook to run the local mypy instead of installing in a new virtualenv

File: mypy.ini (hunk 1 above)
Comment (incognitojam) [reply mypy.ini:6]: Actually might not need this after we add the types

Comment (incognitojam) [reply mypy.ini:6]: nvm, there are lots of packages missing types, will have to do another PR in future to add more types packages

File: .pylintrc [hunk 3]
```diff
@@ -54,7 +54,7 @@ confidence=
 # --enable=similarities". If you want to run only the classes checker, but have
 # no Warning level messages displayed, use"--disable=all --enable=classes
 # --disable=W"
-disable=C,R,W0613,W0511,W0212,W0201,W0106,W0603,W0621,W0703,W1201,W1203,E1136,W1514
+disable=C,R,W0613,W0511,W0212,W0201,W0106,W0603,W0621,W0703,W1201,W1203,E1136,W1514,E1224
```
Comment (incognitojam) [inline .pylintrc:57]: pylint started complaining about unnecessary keyword args (e.g. `video_link=`  in car docs)

File: .pylintrc [hunk 4]
```diff
@@ -54,7 +54,7 @@ confidence=
 # --enable=similarities". If you want to run only the classes checker, but have
 # no Warning level messages displayed, use"--disable=all --enable=classes
 # --disable=W"
-disable=C,R,W0613,W0511,W0212,W0201,W0106,W0603,W0621,W0703,W1201,W1203,E1136,W1514
+disable=C,R,W0613,W0511,W0212,W0201,W0106,W0603,W0621,W0703,W1201,W1203,E1136,W1514,E1124
```
Comment (incognitojam) [inline .pylintrc:57]: pylint started complaining about unnecessary keyword args (e.g. `video_link=` in car docs)

Comment (adeebshihadeh) [reply .pylintrc:57]: these should just be fixed

File: mypy.ini (hunk 1 above)
Comment (adeebshihadeh) [reply mypy.ini:8]: kk, but would like to add those soon. that's free coverage

File: selfdrive/debug/check_freq.py [hunk 5]
```diff
@@ -43,7 +43,7 @@
     if t - prev_print > 1:
       print()
       for name in socket_names:
-        dts = np.diff(rcv_times[name])
+        dts = np.diff(rcv_times[name])  # type: ignore
```
Comment (adeebshihadeh) [inline selfdrive/debug/check_freq.py:46]: why is this needed?

File: selfdrive/debug/check_timings.py [hunk 6]
```diff
@@ -20,6 +20,6 @@
         ts[s].append(m.logMonoTime / 1e6)
 
       if len(ts[s]):
-        d = np.diff(ts[s])
+        d = np.diff(ts[s])  # type: ignore
```
Comment (adeebshihadeh) [inline selfdrive/debug/check_timings.py:23]: and this one?

File: selfdrive/debug/check_freq.py (hunk 5 above)
Comment (incognitojam) [reply selfdrive/debug/check_freq.py]: I haven't figured out why, but now mypy doesn't think `Deque[float]` is a `_SupportsArray`, but `MutableSequence` works fine

File: .pylintrc (hunk 4 above)
Comment (incognitojam) [reply .pylintrc:57]: #25899

File: mypy.ini [hunk 7]
```diff
@@ -1,4 +1,16 @@
...
 python_version = 3.8
-ignore_missing_imports = True
 plugins = numpy.typing.mypy_plugin
+files = body, common, docs, scripts, selfdrive, site_scons, system, tools
```
Comment (adeebshihadeh) [inline mypy.ini:4]: does this work? docs say it's supposed to be actual file paths, but glob patters are supported

**Review by adeebshihadeh — APPROVED** [review]
LGTM once the pylint exception addition is reverted and the files thing is checked out

File: mypy.ini (hunk 7 above)
Comment (incognitojam) [reply mypy.ini:4]: Seems to work fine!
```
((openpilot) ) batman@z840-openpilot4:~/openpilot$ mypy .
Success: no issues found in 469 source files
((openpilot) ) batman@z840-openpilot4:~/openpilot$ mypy
Success: no issues found in 469 source files
```
